src/sulogin-shell/sulogin-shell.c

```diff
@@ -0,0 +1,105 @@
+/***
+  This file is part of systemd.
+
+  Copyright 2017 Felipe Sateler
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+***/
+
+#include <errno.h>
+#include <sys/prctl.h>
+
+#include "bus-util.h"
+#include "log.h"
+#include "process-util.h"
+#include "sd-bus.h"
+#include "bus-error.h"
+#include "signal-util.h"
+
+static bool start_default_target(void) {
```

Comment:
nitpick: we try to always follow the same semantics when propagating errors, and that's by returning negative errno-style errors from functions. Hence, please follow the same logic here, and return an "int" here, with negative errors. This doesn't matter too much, but it's kinda nice to have the whole codebase stick to the same logic here.

(also see CODING_STYLE about this)

Reply:
I figured that if the only caller only cares about the success/failure, a bool would be appropriate. After all, changing to return int just moves the `r < 0` check from this function to the caller, because the caller won't do anything about it.
Anyway, I changed it to return int.